website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+
+};
+
+var LogWrap = {
+    controller: function(args){
+        var self = this;
+        self.userId = args.userId;
+        self.activityLogs = m.prop();
+        self.eventFilter = false;
+        self.dateEnd = moment.utc();
+        self.dateBegin = moment.utc();
+        self.today = moment.utc();
+        self.sixMonthsAgo = moment.utc().subtract(6, 'months');
+        self.page = 1;
+        self.cache = [];
+        self.loading = false;
+        self.div = 8.64e+7;
+        self.canvasHeight = 40;
+        self.totalEvents = 0;
+        self.eventNumbers = {};
+
+        self.getLogs = function(init, reset, update) {
+            if (!(init || reset || update)  && self.cache[self.page - 1]){
+                self.activityLogs(self.cache[self.page - 1]);
+                if (!self.cache[self.page] && self.page < self.lastPage){
+                    self.page = self.page + 1;
+                    self.getLogs(false, false, true);
+                    self.page = self.page - 1;
+                }
+                return;
+            }
+            var query = {
+                'embed': ['nodes', 'user', 'linked_node', 'template_node'],
+                'page': ((self.page/2) | 0) + 1,
+                'page[size]': 20
+            };
+            if (self.eventFilter) {
+                query['filter[action]'] = self.eventFilter;
+            }
+            if (reset || init) {
+                query.aggregates = 1;
+            }
+            if (!init) {
+                query['filter[date][lte]'] = self.dateEnd.toISOString();
+                query['filter[date][gte]'] = self.dateBegin.toISOString();
+            }
+            var url = $osf.apiV2Url('users/' + self.userId + '/logs/', { query : query});
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig, background: (update ? true : false)});
+            promise.then(function(result){
+                self.loading = false;
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                });
+                if (init) {
+                    self.lastDay = moment.utc(result.data[0].attributes.date);
+                    self.dateEnd = self.lastDay;
+                    var firstDay = moment.utc(result.links.meta.last_log_date);
+                    self.firstDay = ((firstDay >= self.sixMonthsAgo) ? firstDay : self.sixMonthsAgo).startOf('month');
+                    var dateBegin = moment.utc(result.data[0].attributes.date).subtract(1, 'months');
+                    self.dateBegin = ((dateBegin > self.firstDay) ? dateBegin : self.firstDay).startOf('day');
+                    if ((self.today - self.firstDay)/2629746000 < 1){
```

Comment:
This number is one month in milliseconds? It's better to make it a constant variable. The increments in this case will not be exactly a month. It's not a big deal but it might become problematic over long term. Ideally it's best to to increments in months.